### ISSUE TITLE
PPO2: policy entropy grows over time

### ISSUE THREAD

**Issue by blobsimulator** [openai/baselines#371]
When training my model with PPO2, my policy entropy grows over time. I can't really figure out how to stop it from growing. The only clear fix I have found is to set `ent_coef = large negative number`, but that just seems wrong to me.

This is my call to PPO2:

```
runner = learn(policy=MlpPolicy, env=env, nsteps=1024, nminibatches=4,
        lam=1, gamma=0.99, noptepochs=3, log_interval=1,
        ent_coef=0.01,
        lr=lambda f : f * 2.5e-4,
        cliprange=lambda f : f * 0.2,
        total_timesteps=2000000)
```

The environment is rather complex and changes over time. Would that affect how the model increases its entropy?

Could it be that I need to run the algorithm for a longer time before the entropy starts dropping?

**Comment by unixpickle** [comment]
It's hard to diagnose the problem without more information about the environment. Here's some general pointers for debugging this kind of explosion:

 * How large are the rewards, and are rewards normalized?
 * Does the problem still happen if you use a much larger batch size?
 * How does the environment treat incoming actions, e.g.
   * Does it clip actions if they are too large?
   * Does it penalize or reward large actions?

It is sometimes difficult to get RL algorithms to learn a good set of action standard deviations. If possible, I'd recommend trying to discretize the action space.

**Comment by scotthuang1989** [comment]
I have the same issue. 

- my reward is between -1 and 1. I don't normalize it, because I don't know the distribution of it. it depend on the action and state.

- I increase batch size to 256 (original batch size is 32), nothing changes. 

- I clip my action to a range.

- I don't reward or penalize large actions. but large action will come with large reward (positive or negtive).

@unixpickle when you suggest discretize action space, you mean use   DQN or  AC + discrete action. I am new to this,  according to knowledge , DQN usually apply to discrete action space, PG apply to continous action space.

Here is the plot of my policy entropy: 


**another thing confuse me is how baseline calculate entropy.
`tf.reduce_sum(self.logstd + .5 * np.log(2.0 * np.pi * np.e), axis=-1)`**


![image](https://user-images.githubusercontent.com/5325686/41269154-4c032750-6e36-11e8-8a8d-34296b900141.png)
